x-pack/plugins/security_solution/server/usage/collector.ts

```diff
@@ -17,7 +18,8 @@ export interface UsageData {
 
 export async function getInternalSavedObjectsClient(core: CoreSetup) {
   return core.getStartServices().then(async ([coreStart]) => {
-    return coreStart.savedObjects.createInternalRepository();
+    // note: we include the cases hidden types here otherwise we would not be able to query them. If at some point cases is not considered a hidden type this can be removed
+    return coreStart.savedObjects.createInternalRepository(SAVED_OBJECT_TYPES);
```

Comment:
NOTE: Cases wasn't working because they are of type "hidden" and you have to expose them here to fix it to work.

Reply:
Thanks for catching this! I noticed a change in the usage collector api that will allow us to not create the so client like this, we could use `soClient` in `CollectorFetchContext`: https://github.com/elastic/kibana/blob/main/src/plugins/usage_collection/server/collector/types.ts#L86

```ts
fetch: async ({ esClient }: CollectorFetchContext): Promise<UsageData> => {
  const internalSavedObjectsClient = await getInternalSavedObjectsClient(core);
  const soClient = internalSavedObjectsClient as unknown as SavedObjectsClientContract;

  return {
    detectionMetrics:
      (await fetchDetectionsMetrics(kibanaIndex, signalsIndex, esClient, soClient, ml)) || {},
   };
},

=>

fetch: async ({ esClient, soClient }: CollectorFetchContext): Promise<UsageData> => {
  const internalSavedObjectsClient = await getInternalSavedObjectsClient(core);

  return {
    detectionMetrics:
      (await fetchDetectionsMetrics(kibanaIndex, signalsIndex, esClient, soClient, ml)) || {},
   };
},

```

Reply:
I tried to swap this code out and we would have to add the cases hidden types into telemetry in order to use it like this. So I am not going to do that work with this PR at the moment.